qa/tasks/mgr/dashboard/test_osd.py

```diff
@@ -51,6 +54,56 @@ def test_scrub(self):
         self._post('/api/osd/0/scrub?deep=True')
         self.assertStatus(200)
 
+    def test_mark_out_and_in(self):
+        self._post('/api/osd/0/mark_out')
+        self.assertStatus(200)
+
+        self._post('/api/osd/0/mark_in')
+        self.assertStatus(200)
+
+    def test_mark_down(self):
+        self._post('/api/osd/0/mark_down')
+        self.assertStatus(200)
+
+    def test_reweight(self):
+        self._post('/api/osd/0/reweight', {'weight': 0.4})
+        self.assertStatus(200)
+
+        sleep(5)
+
+        self._get('/api/osd/0')
+        self.assertAlmostEqual(0.4, self.jsonBody()['osd_map']['weight'], 1)
+
+        self._post('/api/osd/0/reweight', {'weight': 1})
+
+    def test_create_lost_destroy_remove(self):
+        # Create
+        self._post('/api/osd', {
+            'uuid': 'f860ca2e-757d-48ce-b74a-87052cad563f',
+            'svc_id': 5
+        })
+        self.assertStatus(201)
+        # Lost
+        self._post('/api/osd/5/mark_lost')
+        self.assertStatus(200)
+        # Destroy
+        self._post('/api/osd/5/destroy')
+        self.assertStatus(200)
+        # Remove
+        self._post('/api/osd/5/remove')
+        self.assertStatus(200)
+
+    def test_safe_to_destroy(self):
+        self._get('/api/osd/5/safe_to_destroy')
+        self.assertStatus(200)
+        self.assertJsonBody({'safe-to-destroy': True})
+
+        sleep(5)
```

Comment:
Is there no other way than using sleep? Depending on the system hardware and workload this might be not enough and the test fails. In other situations this might slow down the test run if the previous command runs much faster than 5 seconds.